media/js/base/global.js

```diff
@@ -34,15 +34,16 @@ function init_download_links() {
 }
 
 function update_download_text_for_old_fx() {
-    // if using an out of date firefox
-    if (isFirefox() && !isFirefoxUpToDate()) {
-        // look at each button to see if it's set to check for old firefox
-        $('.download-button').each(function() {
-            var $button = $(this);
-
-            if ($button.hasClass('download-button-check-old-fx')) {
+    // look at each button to see if it's set to check for old firefox
+    var $buttons = $('.download-button-check-old-fx');
+
+    // if using Firefox and the page has download buttons
+    if ($buttons.length && isFirefox()) {
```

Comment:
I haven't tested yet on Android, actually. If the API doesn't work on Android for Firefox, it can be another exception to immediately fallback, like Firefox < 35. Let me test on Android now...